docs/spec.md

```diff
@@ -4318,14 +4328,14 @@ and produces a `result` tensor. More formally,
 
 | Label | Name         | Type                                         | Constraints |
 |-------|--------------|----------------------------------------------|-------------|
-| (I1)  | `operand`    | tensor                                       | (C1)        |
-| (I2)  | `dimensions` | 1-dimensional tensor constant of type `si64` | (C2), (C3)  |
+| (I1)  | `operand`    | tensor or per-tensor quantized tensor        | (C1)        |
```

Comment:
For reshape op, it might be better to have a constraint where the same quantization parameters over the given operand and result.

Reply:
The constraint 
```
* (C1) `element_type(operand) = element_type(result)`.
```

ascertains that. PTAL.

Reply:
Could we more precisely define what it means for element types to be equivalent elsewhere in the spec before it's widely replicated? I'm afraid it's not immediately clear to me.

Reply:
Thanks @paulinesho for bringing this up. 

Recently we have re-defined the semantics of `element_type` to be  inclusive for both `TensorType` and `QuantizedTensorType`. Earlier it applies only to `TensorType`. You can find the semantics [here](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#functions-on-types).

In this context, it means
1. The element type of `operand` and `result` should be same. That is, either they are both non-quantized element types or both quantized element type.
2. In the event when `operand` and `result` are quantized element type, then all the components of the `QuantizedElementType` grammar rule, like `storage_type`, `expressed_type`, `scales`, `zero_points` etc., will be the same for `operand` and `result`.

Please let me know if that clarifies. 
Again, sorry @paulinesho and @abattery for the confusion.